Gems/Atom/Tools/AtomToolsFramework/Code/Source/PropertyEditor/ImagePropertyAssetCtrl.cpp

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <PropertyEditor/ImagePropertyAssetCtrl.h>
+
+#include <Atom/RPI.Reflect/Image/AttachmentImageAsset.h>
+#include <Atom/RPI.Reflect/Image/StreamingImageAsset.h>
+
+#include <AzToolsFramework/AssetBrowser/AssetSelectionModel.h>
+#include <UI/PropertyEditor/Model/AssetCompleterModel.h>
+
+namespace AtomToolsFramework
+{
+    using namespace AzToolsFramework;
+
+    ImagePropertyAssetCtrl::ImagePropertyAssetCtrl(QWidget* parent)
+        : PropertyAssetCtrl(parent)
+    {
+    }
+
+    AssetBrowser::AssetSelectionModel ImagePropertyAssetCtrl::GetAssetSelectionModel()
+    {
+        // allows to select two assets which are derieved from ImageAsset
+        AZStd::vector<AZ::Data::AssetType> assetTypes;
+        assetTypes.push_back(azrtti_typeid<AZ::RPI::AttachmentImageAsset>());
+        assetTypes.push_back(azrtti_typeid<AZ::RPI::StreamingImageAsset>());
+        auto selectionModel = AssetSelectionModel::AssetTypesSelection(assetTypes, false);
+
+        selectionModel.SetTitle(m_title);
+        return selectionModel;
+    }
+
+    void ImagePropertyAssetCtrl::ConfigureAutocompleter()
+    {
+        if (m_completerIsConfigured)
+        {
+            return;
+        }
+
+        PropertyAssetCtrl::ConfigureAutocompleter();
+
+        // Use filter from selection model for the auto completer's filter
+        AssetSelectionModel selection = GetAssetSelectionModel();
+        m_model->SetFilter(selection.GetDisplayFilter());
```

Comment:
Can m_model be replaced with the one returned from the function instead of copying the filter?

Reply:
Nope. The m_model is the AssetCompleterModel which is different than the select model